### PR TITLE
feat(#148): add automatic releases and versioning

### PR DESCRIPTION
**Pull request by njuguna-n**
# Description

This PR adds automatic deployment and releases

medic/cht-pipeline#148

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,3 +19,26 @@ Follow the instructions in [the Local CHT Sync Setup documentation](https://docs
 # set environment variables, install dbt dependencies, seed data, run dbt, run test
 ./run_dbt_tests.sh
 ```
+
+## Release Process
+This repo has an automated release process where each feature/bug fix will be released immediately after it is merged to `main`:
+
+1. Update QA with the work to be done to ensure they're informed and can guide development.
```
Comment (andrablaj) [inline README.md:26]: Can we link the [existing documentation](https://docs.communityhealthtoolkit.org/contribute/code/workflow/) about the CHT code workflow to ensure consistency across all the repos? These steps shouldn't be different from the ones stated in the doc.

**Comment by njuguna-n** [comment]
@dianabarsan please review.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20.x
+      - name: Install Semantic Release and plugins
+        run: npm install -g semantic-release @semantic-release/git @semantic-release/changelog @semantic-release/github @semantic-release/commit-analyzer @semantic-release/release-notes-generator
```
Comment (dianabarsan) [inline .github/workflows/release.yml:21]: Would be so cool if these extra dependencies would just get magically installed with npx.

**Review by dianabarsan — APPROVED** [review]
Looking good!

**Pull request merged**